clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -180,32 +180,16 @@ static ConstAtom *analyzeAllocExpr(CallExpr *CE, Constraints &CS,
   return nullptr;
 }
 
-CVarSet ConstraintResolver::getInvalidCastPVCons(Expr *E) {
-  CVarSet Ret;
-  QualType SrcType, DstType;
-  // As getInvalidCastPVCons could be called from non-persistent expressions
-  // we need to explicitly store the generated PVConstraints into persistent
-  // constraints.
-  if (Info.hasPersistentConstraints(E, Context))
-    return Info.getPersistentConstraints(E, Context);
-
-  DstType = E->getType();
-  SrcType = E->getType();
-  if (ImplicitCastExpr *ICE = dyn_cast<ImplicitCastExpr>(E))
-    SrcType = ICE->getSubExpr()->getType();
-
-  if (ExplicitCastExpr *ECE = dyn_cast<ExplicitCastExpr>(E))
-    SrcType = ECE->getSubExpr()->getType();
```

Comment:
I take it that the attempt to look underneath the casts here was misguided, so you dropped them?

Reply:
This function used to take an `Expr` even though it is only ever called on `CastExpr`. Since I changed it to take a `CastExpr`, `getSubExpr()` can be called on the parameter without a `dyn_cast`.

Reply:
Got it.